config/signatures-test.txt

```diff
@@ -1 +1,2 @@
 com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport#verify(com.puppycrawl.tools.checkstyle.api.Configuration, java.lang.String, java.lang.String[]) @ Use inline config parser instead.
+java.nio.file.Files.createTempDirectory(java.lang.String) @ Use of this method is forbidden, please use @TempDir for better temporary directory control
```

Comment:
`java.nio.file.Files#createTempDirectory(java.lang.String)`

Attention to `#`

Reply:
> `java.nio.file.Files#createTempDirectory(java.lang.String)`
> 
> Attention to `#`

I tried with `#` in my local, but the build failed. I was trying to incorporate `createTempFile` aswell as mentioned by rnveach. I have done this: `java.nio.file.Files#createTempDirectory(java.lang.String)`. Let's see.

Reply:
`[ERROR] Failed to execute goal de.thetaphi:forbiddenapis:3.6:testCheck (forbiddenapis-test) on project checkstyle: Parsing signatures failed: Method not found while parsing signature: java.nio.file.Files#createTempDirectory(java.lang.String)`

Still failing.

Reply:
No longer getting parsing error. But, needs to refactor everywhere we have `createTempFile` and `createTempDirectory`. 

```
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.filters.SuppressionFilterTest (SuppressionFilterTest.java:171)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.filters.SuppressionFilterTest (SuppressionFilterTest.java:174)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.filters.SuppressionFilterTest (SuppressionFilterTest.java:203)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.filters.SuppressionFilterTest (SuppressionFilterTest.java:207)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.utils.XpathUtilTest (XpathUtilTest.java:103)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.utils.XpathUtilTest (XpathUtilTest.java:123)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.utils.XpathUtilTest (XpathUtilTest.java:140)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.utils.XpathUtilTest (XpathUtilTest.java:168)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.TreeWalkerTest (TreeWalkerTest.java:223)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.TreeWalkerTest (TreeWalkerTest.java:237)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.TreeWalkerTest (TreeWalkerTest.java:289)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.TreeWalkerTest (TreeWalkerTest.java:548)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.TreeWalkerTest (TreeWalkerTest.java:560)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.TreeWalkerTest (TreeWalkerTest.java:572)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.TreeWalkerTest (TreeWalkerTest.java:587)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.TreeWalkerTest (TreeWalkerTest.java:590)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.TreeWalkerTest (TreeWalkerTest.java:621)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.TreeWalkerTest (TreeWalkerTest.java:623)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.PropertyCacheFileTest (PropertyCacheFileTest.java:95)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.PropertyCacheFileTest (PropertyCacheFileTest.java:152)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.PropertyCacheFileTest (PropertyCacheFileTest.java:172)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.PropertyCacheFileTest (PropertyCacheFileTest.java:201)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.PropertyCacheFileTest (PropertyCacheFileTest.java:202)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.PropertyCacheFileTest (PropertyCacheFileTest.java:229)
[ERROR] Forbidden method invocation: java.nio.file.Files#createTempFile(java.lang.String,java.lang.String,java.nio.file.attribute.FileAttribute[]) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.PropertyCacheFileTest (PropertyCacheFileTest.java:330)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.PropertyCacheFileTest (PropertyCacheFileTest.java:379)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.PropertyCacheFileTest (PropertyCacheFileTest.java:427)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.PropertyCacheFileTest (PropertyCacheFileTest.java:458)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.PropertyCacheFileTest (PropertyCacheFileTest.java:494)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.utils.FilterUtilTest (FilterUtilTest.java:44)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest (AbstractJavadocCheckTest.java:365)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.checks.imports.ImportControlCheckTest (ImportControlCheckTest.java:344)
[ERROR] Forbidden method invocation: java.io.File#createTempFile(java.lang.String,java.lang.String,java.io.File) [Use of this method is forbidden, please use @TempDir for better temporary directory control]
[ERROR]   in com.puppycrawl.tools.checkstyle.checks.imports.ImportControlCheckTest (ImportControlCheckTest.java:347)
[ERROR] Scanned 3799 class file(s) for forbidden API invocations (in 2.51s), 33 error(s).
```

Reply:
Please add all such files to excludes to enforce this rule quickly and refactor code in separate PRs.

The same as we did with other forbidden methods and long migration processes

Reply:
New issue will target to resolve all them file by file.

But, if refactoring is supper easy we can refactor all in this PR :)

Reply:
> New issue will target to resolve all them file by file.
> 
> But, if refactoring is supper easy we can refactor all in this PR :)

I created a new issue: https://github.com/checkstyle/checkstyle/issues/14465. I think refactoring will take some effort, and a new issue might make more sense. Please let me know if i have to do something else from my end.

Reply:
@relentless-pursuit I am approving this, but I am looking to understand why you say there will be some effort in the new issue. The issues you fix in this PR look like a straightforward replace and I am assuming the rest to be the same. There may be some time needed because of the number of files ( 8 ), but it still seems pretty straightforward for each instance in said files.